src/bitcoin/poller/looper.rs

```diff
@@ -46,44 +46,53 @@ fn update_coins(
             ..
         } = utxo;
         // We can only really treat them if we know the derivation index that was used.
-        let address = match address.require_network(network) {
-            Ok(addr) => addr,
-            Err(e) => {
-                log::error!("Invalid network for address: {}", e);
-                continue;
+        let (derivation_index, is_change) = match address {
+            UTxOAddress::Address(address) => {
+                let address = match address.require_network(network) {
+                    Ok(addr) => addr,
+                    Err(e) => {
+                        log::error!("Invalid network for address: {}", e);
+                        continue;
+                    }
+                };
+                if let Some((derivation_index, is_change)) =
+                    db_conn.derivation_index_by_address(&address)
+                {
+                    (derivation_index, is_change)
+                } else {
+                    // TODO: maybe we could try out something here? Like bruteforcing the next 200 indexes?
```

Comment:
This code has been moved but otherwise not modified in this PR so we can plan something for a follow-up. I haven't looked into it in detail yet but I'll create an issue for further discussion.